crates/sui-graphql-rpc/src/types/query.rs

```diff
@@ -615,6 +616,27 @@ impl Query {
             .await
             .extend()
     }
+
+    /// Fetch a list of objects by their IDs and versions.
+    async fn multi_get_objects(
+        &self,
+        ctx: &Context<'_>,
+        object_keys: Vec<ObjectKey>,
+    ) -> Result<Vec<Object>> {
+        let cfg: &ServiceConfig = ctx.data_unchecked();
+        if object_keys.len() > cfg.limits.max_multi_get_objects_keys as usize {
+            return Err(Error::Client(format!(
+                "Number of keys exceeds max limit of '{}'",
+                cfg.limits.max_multi_get_objects_keys
+            ))
+            .into());
+        }
```

Comment:
this check is being duplicated here and in the query limits check. I think it's fine/good to have it done here -- that mirrors what we do for pages, but let's keep it down to one place where we do the check.

Reply:
Transactions IDs does it here in the transaction type, so that's why I added here. I'll keep it here.

Reply:
I do wonder though if it's not better to have it in the query limits checker, isn't that one triggered before this resolver?

Reply:
Yes, it is triggered first, but going back to an earlier point, it's testing for something slightly different: It is bounding the number of output nodes we are going to produce, while this bound is bounding the amount of work we're doing in one DB query. I think it makes sense (and is fine/good) that we are testing them in different places.